lib/performance_data/statistics.rb

```diff
@@ -0,0 +1,216 @@
+    #                        }]
+    #         }
+    #     ]
+    # }
+    #
+    # The method would parse that response into:
+    # [
+    #   {
+    #     total_searches: 180,
+    #     keyword: "login",
+    #     searches: [{
+    #                 timestamp: "2014-09-03T00:00:00Z",
+    #                 value: 104
+    #               }]
+    #   }
+    # ]
+    #
+
+    def search_terms
+      response = data_out.search_terms(slug)
+      stats = []
+
+      if response.raw_response_body != 'null'
+        response = response.to_h
+        response["data"].map do |item|
+          searches = []
+          item["values"].each do |value_item|
+            searches << {
+              timestamp: value_item["_start_at"].to_datetime,
+              value: value_item["searchUniques:sum"]
+            }
+          end
+
+          stats << {
+            total_searches: item["searchUniques:sum"],
+            keyword: item["searchKeyword"],
+            searches: searches
+          }
+        end
+      end
+
+      stats
+    end
+
+    def metrics
+      { lead_metrics: lead_metrics }.tap do |metrics|
+        metrics[:per_page_metrics] = {}
+        part_urls.each do |path|
+          metrics[:per_page_metrics][path] = metrics_for(path)
+        end
+        if metrics[:per_page_metrics] == {}
+          metrics[:per_page_metrics] = nil
+        end
+      end
+    end
+
+    private
+
+    def get_part_urls(content, slug)
+      details = content.fetch("details")
+      return [] unless details.key?("parts")
+
+      parts = details.fetch("parts")
+
+      part_urls = parts.map! do |part|
+        if part.key?("web_url")
+          URI(part["web_url"]).path
+        else
+          "/#{slug}/#{part['slug']}"
+        end
+      end
+      part_urls.unshift(slug.insert(0, "/"))
+    end
+
+    def lead_metrics
+      pageview_data = performance_data_for(:page_views, part_urls)
+      search_data = performance_data_for(:searches, part_urls)
+      problem_data = performance_data_for(:problem_reports, part_urls)
+      search_term_data = performance_data_for(:search_terms, [])
+      if is_multipart
+        return PerformanceData::MultiPartMetrics.new(
+            unique_pageviews: pageview_data,
+            exits_via_search: search_data,
+            problem_reports: problem_data,
+            search_terms: search_term_data
+        )
+      else
+        return PerformanceData::Metrics.new(
+            unique_pageviews: pageview_data,
+            exits_via_search: search_data,
+            problem_reports: problem_data,
+            search_terms: search_term_data
+        )
+      end
+    end
+
+    def metrics_for(path)
+      PerformanceData::Metrics.new(
+          unique_pageviews: performance_data_for(:page_views, [path]),
+          exits_via_search: performance_data_for(:searches, [path]),
+          problem_reports: performance_data_for(:problem_reports, [path])
```

Comment:
Not a problem introduced here, but it's gotta be worth exploring (probably in a separate story/PR) why `search_terms` aren't included in the per page metrics.  It might be that we don't get a break down of search terms for the parts, only for the main page, if that's the case a comment somewhere wouldn't go amiss.